Making the pre-calculated fluid work again

GuillermoCasas opened the pull request:
Now it works fine and is not too expensive, as proven by the T-junction examples. Several bugs concerning the variable management options and some json modifications have been resolved.

GuillermoCasas committed:
eliminate watch

GuillermoCasas committed:
Avoiding adding so many varaibles and adding check for them

I have modified the criteria to add some of the varaibles, since it included by default too manyu variables. In case anything breaks, we can have a look, but this was necessary. I have also added a nice function to check if the model part has that nodal varaible from python

GuillermoCasas committed:
Updating to new json inputs

GuillermoCasas committed:
Solving bug with pressure_gradient recovery

The necessary varaibles were not being added for some of the options. Some pressure gradient recoveries had not been implemented, leading to memory errors

GuillermoCasas committed:
Solving a bug with io of h5py files

The development of these tools had been interrupted at some point and was left not working properly. Now reading and writing works (reading the file just created), although reading is quite slow

GuillermoCasas committed:
Pressure gradient whould by default not be recovered

It is not used and is only possibly relevant to post-processing

GuillermoCasas committed:
Answering a comment

Yes, it is used.

GuillermoCasas committed:
Some more json adaptations

GuillermoCasas committed:
Making the fluid time step coincide witht he load time step

When the fluid was to be calculated one reads it instead of having a separate counter for it. Also some aesthetic changes and comments

maceligueta commented on applications/swimming_DEM_application/python_scripts/swimming_DEM_algorithm.py:
```diff
@@ -442,6 +444,8 @@ def FluidInitialize(self):
         self.fluid_algorithm.ReadFluidModelPart()
+        for node in self.fluid_model_part.Nodes:
+            print(self.fluid_model_part)
 
```
Are you sure you want to print this??

GuillermoCasas replied:
Oops!

GuillermoCasas committed:
Removing print and making termination criterion modular

GuillermoCasas committed:
Using the new termination criterion to avoid seg.fault

The fluid wanted to read further on that what was available (had been precomputed). Now it checks how far it can go and uses that to terminate before if required

maceligueta commented on applications/swimming_DEM_application/python_scripts/variables_management.py:
```diff
@@ -9,6 +9,8 @@ def AddNodalVariables(model_part, variable_list):
     for var in variable_list:
         model_part.AddNodalSolutionStepVariable(var)
+        if model_part.Name == "FluidPart":
+            print(var)
```
This one?

GuillermoCasas replied:
I just removed it

GuillermoCasas committed:
Removing one more print
